$ \\ \textbf{solution of EX 3.1} \\ \\  \text{let 1 kg of product p be x and 1 kg of product q be y} \\ \textbf{problem as linear program} \\ \ min \ \ 1200x+2580y \\ \ 2.18x+1.08333y <=980 \\ \ x \ + \ y >= \ 145 \\ \  x \ >= 105 \\ \ y \ >= 125$

In [2]:
!pip install -q pyomo

     |████████████████████████████████| 9.6 MB 16.7 MB/s 
     |████████████████████████████████| 49 kB 7.1 MB/s 


In [3]:
from pyomo.environ import *

In [4]:
#declare model
model =ConcreteModel()

In [5]:
#declare variable
model.x =Var(domain=Reals)
model.y = Var(domain=Reals)

In [6]:
#objective
model.cost = Objective(expr = 1200*model.x + 2580*model.y,sense = minimize)

In [7]:
#declare constraint
model.constr1 = Constraint( expr = 2.18*model.x + 1.08333*model.y <=980)
model.constr2 = Constraint( expr = model.x + model.y >=145)
model.constr3 = Constraint( expr = model.x >= 105)
model.constr4 = Constraint(expr = model.y >= 125)


In [8]:
!apt-get install -y -qq glpk.utils

Selecting previously unselected package libsuitesparseconfig5:amd64.
(Reading database ... 155673 files and directories currently installed.)
Preparing to unpack .../libsuitesparseconfig5_1%3a5.1.2-2_amd64.deb ...
Unpacking libsuitesparseconfig5:amd64 (1:5.1.2-2) ...
Selecting previously unselected package libamd2:amd64.
Preparing to unpack .../libamd2_1%3a5.1.2-2_amd64.deb ...
Unpacking libamd2:amd64 (1:5.1.2-2) ...
Selecting previously unselected package libcolamd2:amd64.
Preparing to unpack .../libcolamd2_1%3a5.1.2-2_amd64.deb ...
Unpacking libcolamd2:amd64 (1:5.1.2-2) ...
Selecting previously unselected package libglpk40:amd64.
Preparing to unpack .../libglpk40_4.65-1_amd64.deb ...
Unpacking libglpk40:amd64 (4.65-1) ...
Selecting previously unselected package glpk-utils.
Preparing to unpack .../glpk-utils_4.65-1_amd64.deb ...
Unpacking glpk-utils (4.65-1) ...
Setting up libsuitesparseconfig5:amd64 (1:5.1.2-2) ...
Setting up libcolamd2:amd64 (1:5.1.2-2) ...
Setting up libamd2:amd64 

In [9]:
model.pprint()

2 Var Declarations
    x : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    y : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals

1 Objective Declarations
    cost : Size=1, Index=None, Active=True
        Key  : Active : Sense    : Expression
        None :   True : minimize : 1200*x + 2580*y

4 Constraint Declarations
    constr1 : Size=1, Index=None, Active=True
        Key  : Lower : Body               : Upper : Active
        None :  -Inf : 2.18*x + 1.08333*y : 980.0 :   True
    constr2 : Size=1, Index=None, Active=True
        Key  : Lower : Body  : Upper : Active
        None : 145.0 : x + y :  +Inf :   True
    constr3 : Size=1, Index=None, Active=True
        Key  : Lower : Body : Upper : Active
        None : 105.0 :    x :  +Inf :   True
    constr4 : Size=1, Index=None, Active=True
  

In [10]:
SolverFactory('glpk',excutable = '/usr/bin/glpsol').solve(model).write()

# ==========================================================
# = Solver Results                                         =
# ==========================================================
# ----------------------------------------------------------
#   Problem Information
# ----------------------------------------------------------
Problem: 
- Name: unknown
  Lower bound: 448500.0
  Upper bound: 448500.0
  Number of objectives: 1
  Number of constraints: 5
  Number of variables: 3
  Number of nonzeros: 7
  Sense: minimize
# ----------------------------------------------------------
#   Solver Information
# ----------------------------------------------------------
Solver: 
- Status: ok
  Termination condition: optimal
  Statistics: 
    Branch and bound: 
      Number of bounded subproblems: 0
      Number of created subproblems: 0
  Error rc: 0
  Time: 0.01224207878112793
# ----------------------------------------------------------
#   Solution Information
# -------------------------------

In [12]:
#display solution
print('\ncost = ',model.cost())
print('\decision variable')
print('x=',model.x())
print('y = ',model.y())
print('\n constraints')
print('constraint1=',model.constr1())
print('constraint2 = ',model.constr2())
print('constraint3 = ',model.constr3())
print('constraint4 = ',model.constr4())



cost =  448500.0
\decision variable
x= 105.0
y =  125.0

 constraints
constraint1= 364.31624999999997
constraint2 =  230.0
constraint3 =  105.0
constraint4 =  125.0


$\\ \textbf{solution of 3.3} \\ \text{optimal value = 448500} \\ \textbf{value of variable} \\ \ x=105 \\ \ y=125  \\  \\ \\ \textbf{activities of constraint} \\ \text{constraint 1 is inactive} \\ \text{constraint 2 is inactive} \\ \text{constraint 3 is active} \\ \text{constraint 4 is active} $

$ \\ \textbf{solution of 3.4} \\ \text{in our problem initially x>=105 now we got another order of p of 42kg so x >= 105+42 or x >= 147} \\ \text{similar for q initially y >= 125 now become y >= 222}$ 

In [13]:
!pip install -q pyomo

In [14]:
from pyomo.environ import *

In [15]:
#declare model
model2 =ConcreteModel()

In [17]:
#declare variable
model2.x =Var(domain=Reals)
model2.y = Var(domain=Reals)

In [18]:
#objective
model2.cost = Objective(expr = 1200*model2.x + 2580*model2.y,sense = minimize)

In [19]:
#declare constraint
model2.constr1 = Constraint( expr = 2.18*model2.x + 1.08333*model2.y <=980)
model2.constr2 = Constraint( expr = model2.x + model2.y >=145)
model2.constr3 = Constraint( expr = model2.x >= 147)
model2.constr4 = Constraint(expr = model2.y >= 222)

In [20]:
!apt-get install -y -qq glpk.utils

In [21]:
model2.pprint()

2 Var Declarations
    x : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals
    y : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  None :  None : False :  True :  Reals

1 Objective Declarations
    cost : Size=1, Index=None, Active=True
        Key  : Active : Sense    : Expression
        None :   True : minimize : 1200*x + 2580*y

4 Constraint Declarations
    constr1 : Size=1, Index=None, Active=True
        Key  : Lower : Body               : Upper : Active
        None :  -Inf : 2.18*x + 1.08333*y : 980.0 :   True
    constr2 : Size=1, Index=None, Active=True
        Key  : Lower : Body  : Upper : Active
        None : 145.0 : x + y :  +Inf :   True
    constr3 : Size=1, Index=None, Active=True
        Key  : Lower : Body : Upper : Active
        None : 147.0 :    x :  +Inf :   True
    constr4 : Size=1, Index=None, Active=True
  

In [22]:
SolverFactory('glpk',excutable = '/usr/bin/glpsol').solve(model2).write()

# ==========================================================
# = Solver Results                                         =
# ==========================================================
# ----------------------------------------------------------
#   Problem Information
# ----------------------------------------------------------
Problem: 
- Name: unknown
  Lower bound: 749160.0
  Upper bound: 749160.0
  Number of objectives: 1
  Number of constraints: 5
  Number of variables: 3
  Number of nonzeros: 7
  Sense: minimize
# ----------------------------------------------------------
#   Solver Information
# ----------------------------------------------------------
Solver: 
- Status: ok
  Termination condition: optimal
  Statistics: 
    Branch and bound: 
      Number of bounded subproblems: 0
      Number of created subproblems: 0
  Error rc: 0
  Time: 0.013033390045166016
# ----------------------------------------------------------
#   Solution Information
# ------------------------------

In [23]:
#display solution
print('\ncost = ',model2.cost())
print('\decision variable')
print('x=',model2.x())
print('y = ',model2.y())
print('\n constraints')
print('constraint1=',model2.constr1())
print('constraint2 = ',model2.constr2())
print('constraint3 = ',model2.constr3())
print('constraint4 = ',model2.constr4())


cost =  749160.0
\decision variable
x= 147.0
y =  222.0

 constraints
constraint1= 560.9592600000001
constraint2 =  369.0
constraint3 =  147.0
constraint4 =  222.0


$ \\ \textbf{solution of 3.6} \\ \\ \textbf{optimal value = 749160} \\ \\ \textbf{optimal value of decision variable} \\ \ x = 147 \\ \ y= 222 \\ \\ \text{constraint 1 and constraint 2 is inactive constraint 3 and constraint 4 is active} \\ \\ \text{jen industries wants to minimise production cost so previous order is good  it doesn't take new order} $